px-map-behavior-vector-layer.es6.js

```diff
@@ -0,0 +1,119 @@
+(function() {
+  'use strict';
+
+  /****************************************************************************
+   * BEHAVIORS
+   ****************************************************************************/
+
+  /* Ensures the behavior namespace is created */
+  window.PxMapBehavior = (window.PxMapBehavior || {});
+
+  /**
+   *
+   * @polymerBehavior PxMapBehavior.VectorLayer
+   */
+  PxMapBehavior.VectorLayerImpl = {
+    properties: {
+      /**
+       * A string that will be contain the geojson:
+       *
+       * @type {String}
+       */
+      data: {
+        type: String,
+        observer: 'shouldUpdateInst'
+      },
+
+      /**
+       * An object that can specify the style
+       *
+       * @type {Object}
+       */
+      featureStyle: {
+        type: Object,
+        value: null
+      },
+
+      /**
+       * A boolean that if set will show a popover containing all the feature properties when the feature is clicked
+       *
+       * @type {Boolean}
+       */
+      showFeatureProperties: {
+        type: Boolean,
+        value: false
+      },
+
+      /**
+       * A boolean that if set will allow a blank vector layer to be created
+       *
+       * @type {Boolean}
+       */
+      blankLayer: {
+        type: Boolean,
+        value: false
+      }
+    },
+
+    canAddInst() {
+      return ((typeof this.data === 'string') && this.data.length) || this.blankLayer;
+    },
+
+    createInst(options) {
+      let featureData;
+      options.data ? featureData = JSON.parse(options.data) : featureData = null;
+
+      let geojsonLayer = L.geoJson(featureData, {
```

Comment:
I tested this by setting the data to just be a simple geoJson line and the styling is still applied.
However passing the style to the constructor doesn't cause any issues and the styles in `pointToLayer()` are still applied, so this can be added.